include/phasar/DataFlow/IfdsIde/Solver/IterativeIDESolver.h

```diff
@@ -0,0 +1,1331 @@
+#ifndef PHASAR_PHASARLLVM_DATAFLOWSOLVER_IFDSIDE_SOLVER_ITERATIVEIDESOLVER_H
+#define PHASAR_PHASARLLVM_DATAFLOWSOLVER_IFDSIDE_SOLVER_ITERATIVEIDESOLVER_H
+
+/******************************************************************************
+ * Copyright (c) 2022 Fabian Schiebel.
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of LICENSE.txt.
+ *
+ * Contributors:
+ *     Fabian Schiebel
+ *****************************************************************************/
+
+#include "phasar/DataFlow/IfdsIde/EdgeFunctions.h"
+#include "phasar/DataFlow/IfdsIde/Solver/Compressor.h"
+#include "phasar/DataFlow/IfdsIde/Solver/EdgeFunctionCache.h"
+#include "phasar/DataFlow/IfdsIde/Solver/FlowEdgeFunctionCacheNG.h"
+#include "phasar/DataFlow/IfdsIde/Solver/FlowFunctionCache.h"
+#include "phasar/DataFlow/IfdsIde/Solver/IdBasedSolverResults.h"
+#include "phasar/DataFlow/IfdsIde/Solver/IterativeIDESolverBase.h"
+#include "phasar/DataFlow/IfdsIde/Solver/IterativeIDESolverResults.h"
+#include "phasar/DataFlow/IfdsIde/Solver/IterativeIDESolverStats.h"
+#include "phasar/DataFlow/IfdsIde/Solver/StaticIDESolverConfig.h"
+#include "phasar/DataFlow/IfdsIde/Solver/WorkListTraits.h"
+#include "phasar/DataFlow/IfdsIde/SolverResults.h"
+#include "phasar/Domain/BinaryDomain.h"
+#include "phasar/Utils/ByRef.h"
+#include "phasar/Utils/EmptyBaseOptimizationUtils.h"
+#include "phasar/Utils/Logger.h"
+#include "phasar/Utils/Printer.h"
+#include "phasar/Utils/StableVector.h"
+#include "phasar/Utils/TableWrappers.h"
+#include "phasar/Utils/TypeTraits.h"
+
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/ADT/DenseMapInfo.h"
+#include "llvm/ADT/DenseSet.h"
+#include "llvm/ADT/Hashing.h"
+#include "llvm/ADT/STLExtras.h"
+#include "llvm/ADT/SmallBitVector.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/Twine.h"
+#include "llvm/Support/Compiler.h"
+#include "llvm/Support/ErrorHandling.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include <algorithm>
+#include <atomic>
+#include <chrono>
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <type_traits>
+
+namespace psr {
+
+template <typename ProblemTy,
+          typename StaticSolverConfigTy = DefaultIDESolverConfig<ProblemTy>>
+class IterativeIDESolver
```

Comment:
How much of the solver is a duplicate of the other one? Or what is the plan to continue, because keeping the old one and adding the new one seems like quite a maintenance burden without sharing.

Reply:
They do not really share any common code. This is definitely a maintenance issue for the future tthat we should solve eventually. Added an issue for that (#735)